### PR TITLE
Bugfix 6847/fix to show all target words in SP and word bank and better invalidation checking in WA

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- fixed removeMarker to work better with deeply nested spans and broken USFM. Will now show all target words in SP and word bank, and better invalidation checking in WA

#### Please include detailed Test instructions for your pull request:
- checkout tC branch `bugfix-mcleanb-6847`, run `npm run load-apps && npm ci && npm start`
- use github build attached to https://github.com/unfoldingWord/translationCore/pull/6966
- open project at https://drive.google.com/open?id=1nkSWNnvqHw_QVDaU5dIzNf9GZetrEB4Y
- when you open project, should see 38 new invalidations for WA
- open WA and navigate to luke 4:19
- the old builds would not show invalidations until project was exported, and would only show `4:19 और का प्रचार करूँ।”` in SP and only 4 words in the word bank
- but in this build 4:19 should show invalidated icon.  Also you should see many more words in word bank and SP as:  
Screen Shot 2020-06-04 at 10 07 47 AM copy

This change is [Reviewable](https://reviewable.io/reviews/unfoldingword/tc-ui-toolkit/269)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tc-ui-toolkit",
-  "version": "3.0.0",
+  "version": "3.0.1-alpha",
```
Comment (mannycolon) [inline package.json:3]: Shouldn't the alpha tag be removed?

File: package.json [hunk 2]
```diff
@@ -65,7 +65,8 @@
     "react": "^16.0.0",
     "react-dom": "^16.0.0",
     "string-punctuation-tokenizer": "^2.0.0",
-    "word-aligner": "^0.3.0"
+    "word-aligner": "^0.3.0",
+    "usfm-js": "^2.1.0"
```
Comment (mannycolon) [inline package.json:69]: Word-aligner would need to be updated right ?

Comment (PhotoNomad0) [reply package.json:69]: it's a peer dependency, but I can do that once word-aligner is published.

File: package.json (hunk 1 above)
Comment (PhotoNomad0) [reply package.json]: correct

**Pull request merged**